src/neml2/user_tensors/MultiColumnCSVScalar.cxx

```diff
@@ -49,8 +48,10 @@ MultiColumnCSVScalar::expected_options()
   options.set<std::vector<unsigned int>>("column_indices") = {};
   options.set("column_indices").doc() = "Indices of CSV columns.";
 
-  // options.set<char>("delimiter") = ',';
-  // options.set("delimiter").doc() = "CSV delimiter. The default is ','.";
+  EnumSelection delimiter_selection({"COMMA", "SEMICOLON", "SPACE", "TAB"}, "COMMA");
```

Comment:
I think you can do something like this
```cpp
EnumSelection delimiter_selection({"COMMA", "SEMICOLON", "SPACE", "TAB"}, {',', ';', ' ', '\t'}, "COMMA");
```
Later on, when you get the enum selection, you can directly cast it to a character without if else, i.e.
```cpp
auto delim = options.get<EnumSelection>("delimiter").as<char>();
```